### DISCUSSION TITLE
Request for flatpak

### DISCUSSION THREAD

**Discussion by ploum** [espanso/espanso#1190] [Feature requests and Ideas]
Really nice project and definitely what I was looking for.

I'm opening a request for a flatpak packages (which could be published on flathub) as an alternative to Snap.

Thanks !

**Comment by federico-terzi** [comment]
Hey @ploum,

Thank you for the feedback! I'll make sure to investigate it further. Do you have any experience with flatpak packages?

Cheers :)

**Comment by ploum** [comment]
Unfortunately, I tried and was very bad at it. In short, it looks like the only thing to do is to create a manifest (and some administrative tasks with github).

But that manifest is quite hard to write for people like me who hate packaging. YMMV.

https://docs.flatpak.org/en/latest/first-build.html

The good point is that, if I understand it correctly, once the manifest is done, you can forget about it and everything will be automated (or nearly).

**Comment by federico-terzi** [comment]
All right, no problem :) I'm not very familiar with flatpak, I'm wondering whether it sandboxes applications as snap does

**Comment by ploum** [comment]
AFAIK, sandboxing is on by default but not mandatory. A flatpak can have external permission granted. Also, a flatpak can rely on an existing infrastructure (such as the GNOME or KDE desktop)

**Comment by federico-terzi** [comment]
All right, I'll make sure to document myself and get back to you once I know something :)

**Reply by thorrrr** [reply]
Do we have any update on Flatpak being made available ?

**Comment by dddw** [comment]
Flatpak integration would be great to have. 
Especially for Fedora Silverblue users, which has issues with the creation of the /snap folder.
There is a small workaround for it, bu you cannot permantly snap packages without breaking(and having to reisntall them after reboot.

**Comment by reesericci** [comment]
Flatpak is winning the universal packaging wars, and it is the main platform for immutable distros like Silverblue and the new SteamOS

**Comment by reesericci** [comment]
Hello! any updates on this? I can't use espanso w/o it because I run Fedora Silverblue.

--reese

**Comment by srbaker** [comment]
Hello, wondering if anyone's working on Flatpak packaging?  I'd be happy to work on it, if nobody else is, or collaborate if someone is.

I need this because I use Fedora Silverblue.

Cheers!

**Reply by dddw** [reply]
no work being done asik, issue has been open for 3 years. 
Have no experience in flatpak alas

**Reply by srbaker** [reply]
I will take a look.

**Reply by ShawkMusic** [reply]
@srbaker any progress?

**Comment by Janov911** [comment]
Fixing espanso so it can be exported from a distrobox / toolbox would already make a huge difference, would keep it to 1 distro compatibility (like a flatpak) and 1 source version. 
but I guess we’ll have to keep building it for each system.

**Comment by bmp** [comment]
I am on Fedora Silverblue Wayland and would love a flatpak version. 

@srbaker & @reesericci  Currently, I have it working in the following manner,
1. Follow instructions at https://espanso.org/docs/install/linux/#install-on-wayland to compile from source within a distrobox.
2. Export the binary using a distrobox based on instructions at https://github.com/89luca89/distrobox/blob/main/docs/usage/distrobox-export.md
3. Service is registered under systemd
4. I run the service as `espanso start --unmanaged` as can't start the service with `espanso start`:
```
Warning: The unit file, source configuration file or drop-ins of espanso.service changed on disk. Run 'systemctl --user daemon-reload' to reload units.

unable to start service: timed out
Hint: sometimes this happens because another Espanso process is left running for some reason.
      Please try running 'espanso restart' or manually killing all Espanso processes, then try again.
```
But I have no such process running either.

If any of you have a better solution, please do share.

**Comment by Jopp-gh** [comment]
+1 for this request. Currently I'm using AppImage, but would like to have Espanso as a flatpak,  definitely.

**Comment by ghost** [comment]


**Reply by rugk** [reply]
Please just upvote/ :+1: the first post/initial request instead of commenting and pinging everyone.
